tracing_internal_test.go

```diff
@@ -86,6 +88,41 @@ func TestSpanString(t *testing.T) {
 	assert.Equal(t, "TraceID=f,ParentID=0,SpanID=0", span.String())
 }
 
+func TestSetPeerHostPort(t *testing.T) {
+	tracer := mocktracer.New()
+
+	span := tracer.StartSpan("x")
+	setPeerHostPort(span, "localhost:123")
+	span.Finish()
+	rawSpan := tracer.FinishedSpans()[0]
+	assert.Equal(t, uint32(127<<24|1), rawSpan.Tag(string(ext.PeerHostIPv4)))
+	assert.Equal(t, uint16(123), rawSpan.Tag(string(ext.PeerPort)))
+
+	span = tracer.StartSpan("x")
+	setPeerHostPort(span, "adhoc123:bad-port")
+	span.Finish()
+	rawSpan = tracer.FinishedSpans()[1]
+	assert.Equal(t, "adhoc123", rawSpan.Tag(string(ext.PeerHostname)))
+	assert.Nil(t, rawSpan.Tag(string(ext.PeerPort)))
+
+	span = tracer.StartSpan("x")
+	setPeerHostPort(span, "10.20.30.40:321")
+	span.Finish()
+	rawSpan = tracer.FinishedSpans()[2]
+	ip := (((10<<8)|20)<<8|30)<<8 | 40
+	assert.Equal(t, uint32(ip), rawSpan.Tag(string(ext.PeerHostIPv4)))
+	assert.Equal(t, uint16(321), rawSpan.Tag(string(ext.PeerPort)))
+
+	span = tracer.StartSpan("x")
+	ipv6 := []byte{1, 2, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 15, 16}
+	assert.Equal(t, 16, len(ipv6))
+	setPeerHostPort(span, fmt.Sprintf("[%s]:789", net.IP(ipv6)))
```

Comment:
nit: use `%v` instead of `%s` in the format string. `%s` should only be used for strings / byte slices